src/components/Nav/Nav.tsx

```diff
@@ -60,47 +73,84 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
     return this._selectedKey;
   }
 
-  private _renderLink(link: INavLink, linkIndex: number, nestingLevel: number, hasGroupButton: boolean): React.ReactElement<{}> {
+  private _renderAnchorLink(link: INavLink, linkIndex: number, nestingLevel: number): React.ReactElement<{}> {
     let { onLinkClick } = this.props;
 
     // Determine the appropriate padding to add before this link.
     // In RTL, the "before" padding will go on the right instead of the left.
     const isRtl: boolean = getRTL();
-    const paddingBefore: string = (_indentationSize * nestingLevel + (hasGroupButton ? 40 : 20)).toString(10) + 'px';
-
-    const isLinkSelected: boolean = _isLinkSelected(link, this._selectedKey);
-    if (isLinkSelected) {
-      this._selectedKey = link.key ? link.key : undefined;
-    }
+    const paddingBefore: string = (_indentationSize * nestingLevel +
+      (this._hasExpandButton ? _indentWithExpandButton : _indentNoExpandButton)).toString(10) + 'px';
 
     return (
-      <li role='listitem' key={ linkIndex }>
         <a
-          className={ css('ms-Nav-link', { 'is-selected' : isLinkSelected }) }
+          className={ css('ms-Nav-link', nestingLevel > 0 ? 'ms-Nav-SublinkTextSize' : '') }
           style={ { [isRtl ? 'paddingRight' : 'paddingLeft'] : paddingBefore } }
           href={ link.url || 'javascript:' }
           onClick={ onLinkClick }
           aria-label={ link.ariaLabel }
-          title={ link.title ? link.title : '' }
-          target={ link.target || '' }
+          title={ link.title ? link.title : link.name }
+          target={ link.target }
         >
-          { (link.iconClassName ?
+          { link.iconClassName ?
           <i className={ css('ms-Icon', 'ms-Nav-IconLink', link.iconClassName) }></i>
-          : '') }
+          : '' }
          { this.props.onRenderLink(link)}
         </a>
-        { this._renderLinks(link.links, nestingLevel + 1, hasGroupButton) }
-    </li>
     );
   }
 
-  private _renderLinks(links: INavLink[], nestingLevel: number, hasGroupButton: boolean): React.ReactElement<{}> {
+  private _renderButtonLink(link: INavLink, linkIndex: number) {
+    return (
+      <Button
+          className={ css('ms-Nav-link ms-Nav-linkButton', link.iconClassName) }
+          buttonType={ ButtonType.command }
+          icon='Pencil'
+          description={ link.title || link.name }
+          onClick={ link.onClick }>
+          { link.name }
+      </Button>);
+  }
+
+  private _renderCompositeLink(link: INavLink, linkIndex: number, nestingLevel: number): React.ReactElement<{}> {
+    const isLinkSelected: boolean = _isLinkSelected(link, this._selectedKey);
+    if (isLinkSelected) {
```

Comment:
Since I am OOF next week and all this code tested.  I may not split it now but I will consider it next round.  thanks